### PR TITLE
Representation change for `task.exception`

### PR DESCRIPTION
**Pull request by valayDave**
- Added Repr to fix error representation problem
- Added a test case to ensure this code works as expected. 
- Based on #677

### CODE REVIEW

File: test/core/tests/task_exception.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class TaskExceptionTest(MetaflowTest):
+    """
+    A test to validate if exceptions are stored and retrieved correctly
+    """
+    PRIORITY = 1
+    SHOULD_FAIL = True
+
+    @steps(0, ['singleton-end'],required=True)
+    def step_start(self):
+        raise KeyError('Something has went wrong')
```
Comment (savingoyal) [inline test/core/tests/task_exception.py:14]: Minor nit - `Something has gone wrong`

File: test/core/tests/task_exception.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class TaskExceptionTest(MetaflowTest):
+    """
+    A test to validate if exceptions are stored and retrieved correctly
+    """
+    PRIORITY = 1
+    SHOULD_FAIL = True
+
+    @steps(0, ['singleton-end'],required=True)
```
Comment (savingoyal) [inline test/core/tests/task_exception.py:12]: minor nit - white space `], r`

**Review by savingoyal — CHANGES_REQUESTED** [review]

File: metaflow/exception.py [hunk 3]
```diff
@@ -30,6 +30,9 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__ = state
+    
+    def __repr__(self):
+        return str(self)
```
Comment (lsimoneau) [inline metaflow/exception.py:35]: I think `str` returns the whole backtrace, is that desirable here?

Comment (valayDave) [reply metaflow/exception.py:35]: @savingoyal What are your thoughts?

Comment (valayDave) [reply metaflow/exception.py:35]: We can also use `self.type` for `__repr__`;

**Pull request merged**